Fix BNF of CIRCLE, POLYGON and BOX

gmantele opened the pull request:
The idea is to update the BNF grammar so that reflecting the updated description of BOX (even if deprecated in ADQL-2.1), CIRCLE and POLYGON.

The updated grammar parts are about:
- the now optional coord. sys. argument
- the arguments giving a position can now be either pairs of coordinates or point values

Fixes #29

gmantele committed:
Fix BNF of CIRCLE, POLYGON and BOX
(optional coord. sys. + points vs coordinates)

gmantele commented:
_This PR includes the BNF update of the `BOX` function though it is deprecated.
In function of the decision taken in #29 , this PR may be modified so that reverting the fix on `BOX`._

msdemlei reviewed (dismissed):
This should do what it's supposed to do and matches the grammar DaCHS has been using for a few year now.  Let's merge.

The clarification that coord_sys needs to be a literal comes in a separate PR then?

gmantele commented:
I guess we could also use this PR for that. After all, it is mostly related to CIRCLE and POLYGON.
So, you're right, I'll add a commit to apply this clarification in there.

gmantele committed:
Apply ADQL-2.0 Erratum-3 :
the coordinate system argument of `POINT`, `BOX`,
`CIRCLE` and `POLYGON` MUST be a string literal.

gmantele commented:
_The Erratum-3 is now applied (see 3283c17)._

msdemlei reviewed (approved):
Given I've had my hands in this, I may not be the most objective reviewer, but if nobody else steps forward: I'm now happy with all of this.

molinaro-m commented:
Considering the roles in play, I can enforce Markus's approval to enable merge or, @Zarquan can do that with more insight w.r.t. myself.

Zarquan reviewed (approved):
+1 all looks good to me.
